src/main/java/org/janelia/saalfeldlab/n5/codec/RawBlockCodecs.java

```diff
@@ -0,0 +1,124 @@
+	public static final RawDataBlockCodecFactory<double[]> DOUBLE = (byteOrder, blockSize, codec) -> new RawDataBlockCodec<>(DataCodec.DOUBLE(byteOrder), DoubleArrayDataBlock::new, blockSize, codec);
+	public static final RawDataBlockCodecFactory<String[]> STRING =(byteOrder, blockSize, codec) -> new RawDataBlockCodec<>(DataCodec.ZARR_STRING, StringDataBlock::new,  blockSize, codec);
+	public static final RawDataBlockCodecFactory<byte[]> OBJECT =(byteOrder, blockSize, codec) -> new RawDataBlockCodec<>(DataCodec.OBJECT, ByteArrayDataBlock::new, blockSize, codec);
+
+	private RawBlockCodecs() {}
+
+	public static <T> DataBlockCodec<T> createDataBlockCodec(
+			final DataType dataType,
+			final ByteOrder byteOrder,
+			final int[] blockSize,
+			final Codec.BytesCodec codecs) {
+		final RawDataBlockCodecFactory<?> factory;
+		switch (dataType) {
+		case UINT8:
+		case INT8:
+			factory = RawBlockCodecs.BYTE;
+			break;
+		case UINT16:
+		case INT16:
+			factory = RawBlockCodecs.SHORT;
+			break;
+		case UINT32:
+		case INT32:
+			factory = RawBlockCodecs.INT;
+			break;
+		case UINT64:
+		case INT64:
+			factory = RawBlockCodecs.LONG;
+			break;
+		case FLOAT32:
+			factory = RawBlockCodecs.FLOAT;
+			break;
+		case FLOAT64:
+			factory = RawBlockCodecs.DOUBLE;
+			break;
+		case STRING:
+			factory = RawBlockCodecs.STRING;
+			break;
+		default:
+			throw new IllegalArgumentException("Unsupported data type: " + dataType);
+		}
+		final RawDataBlockCodecFactory<T> tFactory = (RawDataBlockCodecFactory<T>)factory;
+		return tFactory.createDataBlockCodec(byteOrder, blockSize, codecs);
+	}
+
+	public interface RawDataBlockCodecFactory<T> {
+
+		DataBlockCodec<T> createDataBlockCodec(ByteOrder byteOrder, int[] blockSize, Codec.BytesCodec codecs);
+	}
+
+	private static class RawDataBlockCodec<T> extends N5Codecs.AbstractDataBlockCodec<T> {
+
+		private final int[] blockSize;
+
+		RawDataBlockCodec(
+				final DataCodec<T> dataCodec,
+				final DataBlock.DataBlockFactory<T> dataBlockFactory,
+				final int[] blockSize,
+				final Codec.BytesCodec codec) {
+
+			super(dataCodec, dataBlockFactory, codec);
+			this.blockSize = blockSize;
+		}
+
+		private int numElements() {
+			if (blockSize.length == 0)
+				return 0;
+
+			int elements = 1;
+			for (int dimLen : blockSize) {
+				elements *= dimLen;
+			}
+			return elements;
+		}
+
+		@Override N5Codecs.BlockHeader createBlockHeader(DataBlock<T> dataBlock, ReadData blockData) {
+
+			return null;
+		}
+
+		@Override public ReadData encode(DataBlock<T> dataBlock) {
+
+			return ReadData.from(out -> {
+				final ReadData blockData = dataCodec.serialize(dataBlock.getData());
+				codec.encode(blockData).writeTo(out);
+			});
+		}
+
+		@Override N5Codecs.BlockHeader decodeBlockHeader(InputStream in) throws IOException {
+
+			return null;
```

Comment:
Thoughts on inverting the current behavior of AbstractDataBlockCodec? that is, instead of extending here and returning `null` for header, we remove the `header` fields from the abstract class, and just implement it for the N5BlockCodec specifically?

Reply:
Good point, my preference is to do what you suggest here. I'll update that.